libs/aws/langchain_aws/agents/base.py

```diff
@@ -609,3 +311,219 @@ def _parse_intermediate_steps(
                     return session_id, session_state
 
         return None, None
+
+class BedrockInlineAgentsRunnable(BaseChatModel):
```

Comment:
What was the motivation behind using the `ChatModel` as base rather than a Runnable? 

ChatModel have specific mechanisms, for example `bind_tools` to work with tools, that this implementation will bypass, so this might not be the right interface for the agents.